userspace/libsinsp/filter.cpp

```diff
@@ -2178,4 +2183,41 @@ bool sinsp_evttype_filter::run(sinsp_evt *evt, uint16_t ruleset)
 
 	return false;
 }
+
+void sinsp_evttype_filter::evttypes_for_ruleset(std::vector<bool> &evttypes, uint16_t ruleset)
+{
+	evttypes.assign(PPM_EVENT_MAX+1, false);
+
+	for(uint32_t etype = 0; etype < PPM_EVENT_MAX; etype++)
+	{
+		// Catchall filters (ones that don't explicitly refer
+		// to a type) must run for all event types.
+		for(filter_wrapper *wrap : m_catchall_evttype_filters)
+		{
+			if(wrap->enabled.size() >= (size_t) (ruleset + 1) &&
+			   wrap->enabled[ruleset])
+			{
+				evttypes[etype] = true;
+				break;
+			}
+		}
+
+		if(!evttypes[etype])
+		{
+			list<filter_wrapper *> *filters = m_filter_by_evttype[etype];
+			if(filters)
+			{
+				for(filter_wrapper *wrap : *filters)
+				{
+					if(wrap->enabled.size() >= (size_t) (ruleset + 1) &&
+					   wrap->enabled[ruleset])
+					{
+						evttypes[etype] = true;
+						continue;
```

Comment:
@mstemm this can be a `break` and the entire `for loop` could be extracted in a separate function because it's the same here and at https://github.com/draios/sysdig/pull/958/files#diff-c05de36fec698358ec86e73e5cdb22b2R2195

Reply:
Ok, look at https://github.com/draios/sysdig/pull/964.